src/core/build.jl

```diff
@@ -12,9 +12,20 @@ function _instance_datatype(md::ModelDef, def::DatumDef, start::Int)
     elseif dims[1] != :time
         T = Array{dtype, num_dims}
     
-    else
-        step = step_size(md)
-        T = TimestepArray{dtype, num_dims, start, step}
+    else   
+
+        #T = TimestepArray{AbstractTimestep, dtype, num_dims}
+
+        # LFR-TBD:  There may be a more elegant way to carry out the logic below.  
+        # We need to access years and stepsize, so this might have to do with
+        # how the isuniform function is used etc.
+        years = years_array(md)
```

Comment:
One small improvement would be to test for `isuniform` when the time dimension is set, and store this in the `ModelDef`. Then you could retrieve the years array only when its needed (i.e., for variable timesteps). Same pattern in connections.jl.

Reply:
Good idea, I was thinking that we might be able to put isuniform calls somewhere more convenient.  I think this pattern appears once more in `defs.jl`, so that's three times we could make smoother.

Reply:
Taken care of via discussion today, still should take a look at new code to find places to simplify/optimize